### ISSUE TITLE
Add more border_type options for filter2D

### ISSUE THREAD

**Issue by tlsshh** [kornia/kornia#973]
## 🚀 Feature
1. For filter2D, add border_type options that are provided by **filter2** from **Matlab**.
    For example, 'valid' option:
      ```
      %     'valid' - returns only those parts of the correlation
      %               that are computed without the zero-padded
      %               edges, size(Y) < size(X).
      ```
2. If possible, add a border_type option for some loss functions (e.g. ssim_loss). This option is used to control the behavior of filter2D in ssim_loss.

## Motivation
In my field, some metrics were written by Matlab, and I have to migrate these codes into PyTorch. The issue is, the Matlab code used a border_type option (i.e., "valid") that is not supported by Kornia, resulting in different scores between Matlab-version and PyTorch-version.

## Pitch
filter2D function will support "valid" border type.

## Alternatives
Don't think of any alternative solutions.

**Comment by shijianjian** [comment]
If I understand it right, 'valid' should not be a ```border_type``` here but a padding strategy similar to tensorflow's ```valid padding``` and ```same padding```. Meanwhile, what has been implemented in kornia is the ```same padding```? If everything above is correct, I think the ```valid padding``` could be a good feature to add.

**Comment by edgarriba** [comment]
@tlsshh @shijianjian go for it, please provide a concise api for that

**Comment by shijianjian** [comment]
@tlsshh Are you interested in sending a PR regarding the ```valid``` padding?

**Comment by tlsshh** [comment]
> @tlsshh Are you interested in sending a PR regarding the `valid` padding?

I would like to ... but I am not clear about what I should do, and I never programed anything like that before ...

**Comment by edgarriba** [comment]
@tlsshh join our slack - we'll guide you

**Comment by tlsshh** [comment]
> @tlsshh join our slack - we'll guide you

Ok. I joined the slack. What should I do next?

**Comment by edgarriba** [comment]
join #mentoring channel and let's discuss there
